### PR TITLE
Implemented Header component

### PR DESCRIPTION
**Pull request by yrahul3910**
Completed the following tasks:

* Display a logo in SVG format
* Link logo to home page
* Add "How it works" and "About" links pointing to `#how-it-works` and `#about` respectively
* Add "Search" link pointing to `/search?q=javascript`

### CODE REVIEW

**Comment by mentor-tara** [comment]
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime, consider starting with the next task already to practice the professional Git workflow. [If you need a refresher, feel free to follow this section in the course again.](https://profy.dev/project/reddit-timer/while-waiting)[](ckma5leps3kko0b87rxtjjp24)

**Review by mentor-tara — CHANGES_REQUESTED** [review]
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [ ] Use human readable URLs [:link:](https://profy.dev/project/reddit-timer/review-tips/human-readable-links)
- [ ] Use absolute instead of relative links: `<Link to="/search" />` [:link:](https://profy.dev/project/reddit-timer/review-tips/use-absolute-links)
- [ ] No components that solely copy props to their children [:link:](https://profy.dev/project/reddit-timer/review-tips/unnecessary-components)
- [ ] Optimize svg files with SVGOMG [:link:](https://profy.dev/project/reddit-timer/review-tips/optimize-svg-with-svgomg)
- [ ] Use react-router Link instead of HTML anchor tag [:link:](https://profy.dev/project/reddit-timer/review-tips/link-instead-of-anchor-tag)

**Styles:**
- [ ] **Where to put your styles** [:link:](https://profy.dev/project/reddit-timer/review-tips/where-to-put-styles)
- [ ] **Refrain from styling children with CSS selectors** [:link:](https://profy.dev/project/reddit-timer/review-tips/css-selectors)
- [ ] **Extract shared CSS values** [:link:](https://profy.dev/project/reddit-timer/review-tips/common-css-values-in-theme)
- [ ] Be careful with pixel-perfectness when using relative font sizes like `0.9em` [:link:](https://profy.dev/project/reddit-timer/review-tips/careful-with-relative-font-size)

**Tests:**
- [ ] **Use React Testing Library for testing** [:link:](https://profy.dev/project/reddit-timer/review-tips/use-react-testing-library)
- [ ] Don't write snapshot tests [:link:](https://profy.dev/project/reddit-timer/review-tips/no-snapshot-tests)
- [ ] **Prefer integration tests over unit tests** [:link:](https://profy.dev/project/reddit-timer/review-tips/no-unit-tests)
- [ ] Get the logo by role with `getByRole` [:link:](https://profy.dev/project/reddit-timer/review-tips/get-logo-by-role)
- [ ] How to test hash links [:link:](https://profy.dev/project/reddit-timer/review-tips/testing-hash-links)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/header-ae7e45f0-51b2-433a-b293-3bef0ae80cce). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckmaavvyw3m7p0b358cxn7dzq)

**Review by mentor-tara — APPROVED** [review]
Great work! :tada::tada::tada: You can merge the PR now.[](ckmabg2tk3mkv0b35j06077kr)

File: src/sass/header.sass [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  display: flex
+  justify-content: space-between
+  padding: 32px 80px
+
+  img
+    margin-top: 8px
+
+  .links
+    display: flex
+    gap: 25px
```
Comment (jkettmann) [inline src/sass/header.sass:11]: Nice

File: src/sass/header.sass [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  display: flex
+  justify-content: space-between
+  padding: 32px 80px
+
+  .links
+    display: flex
+    gap: 25px
+
+    a
+      font-family: Montserrat
```
Comment (jkettmann) [inline src/sass/header.sass:14]: This could be defined as the global font

File: src/sass/header.sass (hunk 1 above)
Comment (yrahul3910) [reply src/sass/header.sass:11]: Thank you!!

File: package.json [hunk 3]
```diff
@@ -6,12 +6,14 @@
...
     "@testing-library/user-event": "^12.0.11",
+    "node-sass": "^5.0.0",
     "normalize.css": "^8.0.1",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-router": "^5.2.0",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "^4.0.1"
+    "react-scripts": "^4.0.1",
+    "styled-components": "^5.2.3"
```
Comment (jkettmann) [inline package.json:16]: It doesn't look like you use `styled-components`. So you could remove the dependency as well

File: src/sass/header.sass (hunk 2 above)
Comment (yrahul3910) [reply src/sass/header.sass:14]: Makes sense! Do you mean move this to a Sass variable in my `index.sass`?

Comment (jkettmann) [reply src/sass/header.sass:14]: That file, yes. But not as a variable. Rather style the body so all the text in your app has that family by default

```
body {
  font-family: Montserrat;
}
```

Comment (jkettmann) [reply src/sass/header.sass:14]: Otherwise you have to set the font for every text element 😉

File: src/sass/header.sass (hunk 1 above)
Comment (jkettmann) [reply src/sass/header.sass:11]: Turns out gap isn't widely supported in combination with flex: https://caniuse.com/?search=gap

Comment (yrahul3910) [reply src/sass/header.sass:11]: Huh, that's interesting. I wonder if it's in the works. Surprised Safari and Opera don't.

**Pull request merged**